coalib/parsing/Globbing.py

```diff
@@ -29,6 +29,39 @@ def _end_of_set_index(string, start_index):
     return closing_index
 
 
+def escape_brackets_and_parentheses(string):
+    """
+    Escapes the given string with [c] pattern
+    """
+    # We need to handle '[' and ']' specially
+    tmp = ""
+    prev = 0
+    while True:
+        lb = string[prev:].find("[")
+        rb = string[prev:].find("]")
+        if lb is -1 or rb is -1:
+            cur = max(lb, rb)
+        else:
+            cur = min(lb, rb)
+        if cur is -1:
+            tmp += string[prev:]
+            break
+        cur = cur + prev
+        tmp += string[prev:cur]
+        tmp += "[" + string[cur] + "]"
+        prev = cur + 1
+    return tmp.replace("(", "[(]").replace(")", "[)]")
+
+
+def unescape_brackets_and_parentheses(string):
+    """
+    Unescapes the given string
+    Note: Only for strings escaped with escape_brackets_and_parentheses
+    """
+    string = string.replace("[(]", "(").replace("[)]", ")")
+    return string.replace("[[]", "[").replace("[]]", "]")
```

Comment:
still might happen, any character can be in a path actually (even a slash
that is not a directory seperator is somehow possible)

Sincerely,

Lasse Schuirmann

contact@viperdev.io
http://viperdev.io/

2016-02-29 18:47 GMT+01:00 Adhityaa Chandrasekar <notifications@github.com>:

> In coalib/parsing/Globbing.py
> <https://github.com/coala-analyzer/coala/pull/1644#discussion_r54447430>:
>
> > +            tmp += string[prev:]
> > +            break
> > +        cur = cur + prev
> > +        tmp += string[prev:cur]
> > +        tmp += "[" + string[cur] + "]"
> > +        prev = cur + 1
> > +    return tmp.replace("(", "[(]").replace(")", "[)]")
> > +
> > +
> > +def unescape_brackets_and_parentheses(string):
> > +    """
> > +    Unescapes the given string
> > +    Note: Only for strings escaped with escape_brackets_and_parentheses
> > +    """
> > +    string = string.replace("[(]", "(").replace("[)]", ")")
> > +    return string.replace("[[]", "[").replace("[]]", "]")
>
> This escapes just the paths *before* globs. So there shouldn't be any |
> (unless of course, the user names a directory with (...|...), which is
> super-obscure).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/1644/files#r54447430>.
>